### ISSUE TITLE
S4462 False Positive

### ISSUE THREAD

**Issue by cmenzi** [SonarSource/sonar-dotnet#2794]
### Description

The statement below give a false positive of `S4462`

### Repro steps

Use somewhere in code: `nameof(Task<object>.Result)` 

### Expected behavior

S4462 should not create an warning.

### Actual behavior

A warning S4462 is created.

### Related information

* SonarC# Version: SonarAnalyzer.CSharp 8.0.0.9566
* Visual Studio Version VS2019 16.3.9

**Comment by Corniel** [comment]
Confirmed:

``` C#
using System.Threading;

public class S4462
{
    public override string ToString()
    {
        return nameof(Task<object>.Result); // Compliant, nameof() does not execute asyn code.
    }
}
```
